base/version.jl

```diff
@@ -110,15 +112,15 @@ function VersionNumber(v::AbstractString)
     m = match(VERSION_REGEX, v)
     m === nothing && throw(ArgumentError("invalid version string: $v"))
     major, minor, patch, minus, prerl, plus, build = m.captures
-    major = parse(VInt, major)
+    major = parse(VInt, major::AbstractString)
```

Comment:
I think this can be `SubString{String}` since `match` is only usable with `String` and `SubString{String}`.

Reply:
Oops, that was leftover from an experiment. The important change is the `VerTuple` assert.